### DISCUSSION TITLE
Allow org owner to revoke members' SAML sessions via the API

### DISCUSSION THREAD

**Discussion by burimbd** [community/community#45562] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

**tldr**: Add API endpoints to revoke a member's active SSO session from the Github Organization

## In the web

As an org owner, I can follow the instructions in this [page](https://docs.github.com/en/enterprise-cloud@latest/organizations/granting-access-to-your-organization-with-saml-single-sign-on/viewing-and-managing-a-members-saml-access-to-your-organization#viewing-and-revoking-an-active-saml-session) to revoke the active SAML sessions for my org users. Under the hood, I can see the following calls being made:

`GET https://github.com/orgs/{OrgID}/people/{GithubID}/sso` -> Lists all active sessions for a user.

`POST https://github.com/orgs/{OrgID}/people/{GithubID}/sso_session/{SessionID}`  -> Revokes a specific session for the user.

The member's session with the org is revoked and they would have to sign in again, which is the desired behavior.

## Ask

We would like to have a similar functionality available via the API so that we can revoke a member's session programmatically and have reliable automation in place.

`GET https://api.github.com/orgs/{OrgID}/members/{GithubID}/sso` -> Lists all active sessions for a user.

`POST https://api.github.com/orgs/{OrgID}/members/{GithubID}/sso_session/{SessionID}`  -> Revokes a specific session for the user.

## Use cases

1. Employee goes/is placed on leave -> We can suspend their account at the IDP, but we cannot automatically revoke their GitHub sessions. An owner has to manually go and revoke their Sessions on the Web, which might not happen as quickly as we would like it and potentially cause security issues.
2. Account takeover has occurred on the IDP -> We can reset/suspend corp accounts, but we cannot revoke github sessions. A malicious third party can use the active session to exfiltrate company ip and cause damage in the org.
